internal/functions/efgs/redismutex/redis-mutex.go

```diff
@@ -0,0 +1,90 @@
+package redismutex
+
+import (
+	"context"
+	"fmt"
+	"github.com/covid19cz/erouska-backend/internal/logging"
+	redisclient "github.com/go-redis/redis/v8"
+	"github.com/go-redsync/redsync/v4"
+	"github.com/go-redsync/redsync/v4/redis/goredis/v8"
+	"os"
+	"sync"
+	"time"
+)
+
+var rs *Connection
+var ctx context.Context
+
+type lazyConnection func() *redsync.Redsync
+
+//Connection Contains lazy Redsync connection
+type Connection struct {
+	inner lazyConnection
+}
+
+func init() {
+	ctx = context.Background()
+
+	connect := func() *redsync.Redsync {
+		logger := logging.FromContext(ctx).Named("efgs.redis-mutex.connect")
+
+		logger.Debug("Connecting to EFGS Redis")
+
+		addr, ok := os.LookupEnv("EFGS_REDIS_ADDR")
+		if !ok {
+			panic("EFGS_REDIS_ADDR env missing")
+		}
+
+		client := redisclient.NewClient(&redisclient.Options{
+			Addr: addr,
+			DB:   1, // here it differs from normal Redis client!
```

Comment:
so all this boilerplate is here just because mutexes use db 1 a other use db 0? :))

Reply:
Well the boilerplate here are four rows :wink: I believe reusing would cause more harm than this in this case.

Reply:
the rest is not? ah okay, I missed that:)